### ISSUE TITLE
Mamba2 really work？

### ISSUE THREAD

**Issue by mahao18cm** [state-spaces/mamba#407]
I used to use Selfattention layer. But when i exchanged selfattention with mamba2. Why mamba2 needs more memory? Maybe I don't konw how to use Mamba?
            mamba = mmm2.Mamba2(128)
            #self.encoder_layers = nn.ModuleList([SelfAttentionLayer(cfg.cost_latent_dim, cfg, dropout=cfg.dropout) for idx in range(self.depth)])
            self.encoder_layers = nn.ModuleList([mamba for idx in range(self.depth)])
Anyone could help me ?

**Comment by tridao** [comment]
Huh you're using the same mamba object for all the layers instead of constructing new objects

**Comment by mahao18cm** [comment]
> Huh you're using the same mamba object for all the layers instead of constructing new objects

You mean that i don't need for idx in range(self.depth). I just use self.encoder_layers = mamba? I am a beginner. So my problem maybe foolish.

**Comment by pancodia** [comment]
> nn.ModuleList([mamba for idx in range(self.depth)])

All the elements in the module list points to the same `mamba`. 
You should do something like `nn.ModuleList([mmm2.Mamba2(128) for idx in range(self.depth)])`

**Comment by mahao18cm** [comment]
> > nn.ModuleList([mamba for idx in range(self.depth)])
> 
> All the elements in the module list points to the same `mamba`. You should do something like `nn.ModuleList([mmm2.Mamba2(128) for idx in range(self.depth)])`

Thanks for your advice. But I am not sure that it really works. Because i try to use this part in CV. I hope we can continue contact about using mamba.
